otelgoyek/otelgoyek.go

```diff
@@ -0,0 +1,94 @@
+// Package otelgoyek provides OpenTelemetry instrumentation for goyek.
+package otelgoyek
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/goyek/goyek/v2"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/trace"
+)
+
+const (
+	instrumentationName    = "github.com/goyek/x/otelgoyek"
+	instrumentationVersion = "0.2.0"
+)
+
+// Instrument adds OpenTelemetry tracing instrumentation to the provided flow.
+func Instrument(flow *goyek.Flow, opts ...Option) error {
+	cfg := newConfig(opts)
+
+	tracer := cfg.TracerProvider.Tracer(instrumentationName, trace.WithInstrumentationVersion(instrumentationVersion))
+
+	e := executor{tracer}
+	flow.UseExecutor(e.Middleware)
+
+	r := runner{tracer}
+	flow.Use(r.Middleware)
```

Comment:
We should probably return the middlewares instead of "using" them so that the user can decide when it is the best place to "add" them.

Then the task runner middleware can be registered before others to have "clean" output of the task. And the flow executor middleware can be registered after others to have "full" output of the flow.
And  the flow

Reply:
7d92f47